src/utils/contracts.js

```diff
@@ -160,31 +160,130 @@ export async function setupContracts(account, library, chainId, fetchId) {
     })
   }
 
-  const fetchAPY = async () => {
+  const _rebasingCreditsPerToken = async (block = 'latest') => {
+    if (block === 'latest' && ContractStore.rebasingCreditsPerToken) {
+      return ContractStore.rebasingCreditsPerToken
+    }
+    const rebasingCreditsPerTokenHex = await jsonRpcProvider.call(
+      {
+        to: ContractStore.XUSD,
+        data: '0x6691cb3d', // rebasingCreditsPerToken()
+      },
+      block
+    )
+    const rebasingCreditsPerToken =
+      rebasingCreditsPerTokenHex === '0x'
+        ? '0'
+        : ethers.utils.formatUnits(rebasingCreditsPerTokenHex, 16)
+    if (block === 'latest') {
+      ContractStore.update((s) => {
+        s.rebasingCreditsPerToken = rebasingCreditsPerToken
+      })
+    }
+    return rebasingCreditsPerToken
+  }
+
+  const _rebasingCredits = async (block = 'latest') => {
+    if (block === 'latest' && ContractStore.rebasingCredits) {
+      return ContractStore.rebasingCredits
+    }
+    const rebasingCreditsHex = await jsonRpcProvider.call(
+      {
+        to: ContractStore.XUSD,
+        data: '0x077f22b7', // rebasingCreditsPerToken()
+      },
+      block
+    )
+    const rebasingCredits =
+      rebasingCreditsHex === '0x'
+        ? '0'
+        : ethers.utils.formatUnits(rebasingCreditsHex, 16)
+    if (block === 'latest') {
+      ContractStore.update((s) => {
+        s.rebasingCredits = rebasingCredits
+      })
+    }
+    return rebasingCredits
+  }
+
+  const fetchAPY = async (days = 30) => {
     try {
-      const response = await fetch(process.env.APR_ANALYTICS_ENDPOINT)
-      if (response.ok) {
-        const json = await response.json()
-        const apy = aprToApy(parseFloat(json.apr), 7)
-        ContractStore.update((s) => {
-          s.apy = apy
-        })
+      const isDemoMode = process.env.ETHEREUM_RPC_CHAIN_ID == 43113
+      let current, past
+      const block = await jsonRpcProvider.getBlockNumber()
+
+      if (!isDemoMode) {
+        // FIXME using block.timestamp to get 30 days ago efficiently
+        const pastBlock = block - ((3600 * 24) / 2) * days
```

Comment:
To get the trailing 30-day APY, say, one needs to find the nearest block to 30-days ago https://support.avax.network/en/articles/5106526-measuring-time-in-smart-contracts

(unlike ETH where you can just use 15 seconds/block, AVAX wants you to use the block.timestamp, which is accurate to ~30s)